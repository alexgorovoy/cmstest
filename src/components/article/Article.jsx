import React, { Component } from 'react';
import { ArticleShape } from '../../shapes/ArticleShape';
import './Article.css';

class Article extends Component {
  
  setHTML(content) { 
    return { __html: content };
  }

  render() {
    const { article: {title, body} } = this.props;    
    return (        
        <React.Fragment>
            <h2>{title}</h2>
            <p dangerouslySetInnerHTML={this.setHTML(body)} />
        </React.Fragment>
    );
  }
}

Article.propTypes = {
    article: ArticleShape.isRequired
};

Article.defaultProps = {
  article: {},  
}

export default Article;
