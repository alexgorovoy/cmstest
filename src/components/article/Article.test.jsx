import React from 'react';
import renderer from 'react-test-renderer';
import Article from './Article';

test('Article / renders as expected', () => {
    const article = {
        title: "test title",
        body: "test body <p>with tags</p>"
    };

    const component = renderer.create(
        <Article article={article}/>
      );
      let hero = component.toJSON();
      expect(hero).toMatchSnapshot();    
});