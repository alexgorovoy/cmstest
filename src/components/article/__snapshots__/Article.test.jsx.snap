// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article / renders as expected 1`] = `
Array [
  <h2>
    test title
  </h2>,
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "test body <p>with tags</p>",
      }
    }
  />,
]
`;
